### PR TITLE
request refactored

### PR DESCRIPTION
**Pull request by decobots**
#48

### CODE REVIEW

File: src/request.py [hunk 1]
```diff
@@ -1,46 +1,56 @@
...
-    response = requests.request(method=method_type,
-                                url=url,
-                                params=params,
-                                data=data,
-                                auth=kwargs.get("auth", None))
-    if response.status_code != 200:
-        log.exception(f'response status code !=200 (response status code {response.status_code})')
-        raise ValueError("response status code !=200 ", response.text)
-    try:  # try if response xml, in other case consider that it is json or bytes
+class Request:
```
Comment (livercat) [inline src/request.py]: Rename into something like “Requester”

File: src/request.py [hunk 2]
```diff
@@ -1,46 +1,56 @@
...
+    @staticmethod
+    def _request_basic(method_type: str, url: str, **kwargs) -> requests.Response:
+        log.debug(f"started requester to url '{url}'")
```
Comment (livercat) [inline src/request.py]: “Started request”

File: src/request.py [hunk 3]
```diff
@@ -1,46 +1,56 @@
...
+        params = None
+        data = None
```
Comment (livercat) [inline src/request.py]: You can drop “if” and default None values, just always do both .get’s

File: src/request.py [hunk 4]
```diff
@@ -1,46 +1,56 @@
...
+                                    data=data,
+                                    auth=kwargs.get("auth", None))
+        if response.status_code != 200:
+            log.exception(f'response status code !=200 (response status code {response.status_code})')
+            raise ValueError("response status code !=200 ", response.text)
+        return response
+
+    def request_json(self, method_type: str, url: str, **kwargs) ->Dict:
+        response = self._request_basic(method_type, url, **kwargs)
+        result = response.json()
```
Comment (livercat) [inline src/request.py]: Add custom exception and a test for it

File: src/request.py [hunk 5]
```diff
@@ -1,46 +1,56 @@
...
+        log.debug(f'server returned json')
+        return result
+
+    def request_xml(self, method_type: str, url: str, **kwargs) -> ElementTree:
+        response = self._request_basic(method_type, url, **kwargs)
         result = ElementTree.fromstring(response.text)
```
Comment (livercat) [inline src/request.py]: Add custom exception and a test for it

File: src/request.py (hunk 1 above)
Comment (decobots) [reply src/request.py]: done

File: src/request.py (hunk 2 above)
Comment (decobots) [reply src/request.py]: done

File: src/request.py (hunk 3 above)
Comment (decobots) [reply src/request.py]: done

File: src/request.py (hunk 4 above)
Comment (decobots) [reply src/request.py]: done

File: src/request.py (hunk 5 above)
Comment (decobots) [reply src/request.py]: done

File: src/requester.py [hunk 6]
```diff
@@ -29,15 +25,23 @@ def _request_basic(method_type: str, url: str, **kwargs) -> requests.Response:
             raise ValueError("response status code !=200 ", response.text)
         return response
 
-    def request_json(self, method_type: str, url: str, **kwargs) ->Dict:
+    def request_json(self, method_type: str, url: str, **kwargs) -> Dict:
         response = self._request_basic(method_type, url, **kwargs)
-        result = response.json()
-        log.debug(f'server returned json')
+        try:
+            result = json.loads(response.content)
```
Comment (livercat) [inline src/requester.py]: response.json() was correct

File: tests/test_requester.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+from unittest import mock
+from xml.etree import ElementTree
+
+import pytest
+import requests
+
+from src.requester import Requester
+
+
+def test__request_basic_incorect():
```
Comment (livercat) [inline tests/test_requester.py]: 1) double underscore
2) incorect -> incorrect

File: tests/test_requester.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+def test_request_xml_correct():
+    r = Requester()
+    r._request_basic = mock.MagicMock()
+    r._request_basic.return_value.content = ElementTree.fromstring("""<rsp stat="ok"><photos>
+                                    <photo id="2636" secret="a123456" server="2" title="test_04" farm="5"/>
+                                    </photos></rsp>""")
+    r.request_xml("GET", "URL")
+
+
+def test_request_xml_correct():
```
Comment (livercat) [inline tests/test_requester.py]: correct -> incorrect

File: tests/test_requester.py (hunk 7 above)
Comment (decobots) [reply tests/test_requester.py]: fixed

File: tests/test_requester.py (hunk 8 above)
Comment (decobots) [reply tests/test_requester.py]: fixed

File: src/requester.py (hunk 6 above)
Comment (decobots) [reply src/requester.py]: fixed

**Pull request merged**